src/api/faq/faq.controller.js

```diff
@@ -43,4 +43,42 @@ async function getFAQ(req, res) {
   }
 }
 
-module.exports = { getFAQ };
+async function addFAQ(req, res) {
+  try {
+    const { question, answer } = req.body;
+    if (!question || !answer) throw new ReferenceError('Make sure your request includes a question and answer.');
+    await new FAQ({ question, answer }).save();
+    return res.json({ message: 'FAQ added successfully to DB.' });
+  } catch (error) {
+    const {
+      message,
+    } = error;
+    const status = error instanceof ReferenceError ? 404 : 500;
+
+    return res.status(status).json({
+      status,
+      message,
+    });
+  }
+}
+
+async function removeFAQ(req, res) {
+  try {
+    const { id: _id } = req.params;
+    const deletedFAQ = await FAQ.deleteOne({ _id });
+    if (deletedFAQ.deletedCount === 0) throw new ReferenceError('There is no FAQ to delete with that ID.');
```

Comment:
This is only called if the deletion above was successful, if the promise failes, it is caught by the trycatch.

Reply:
@ewenjo The way mongoose is behaving seems to handle all cases. It could be I am missing something here. When I was testing it, if an invalid ID was requested it wouldn't throw, it would just be successful with a deleted count of 0.

Reply:
It turns out it only throws an error if you try to delete a FAQ with an invalid id, Mongodb validates an id before it tries to use it. Something like "123" or "abc" will be invalid ids and give the output `{"status":500,"message":"Cast to ObjectId failed for value \"123\" at path \"_id\" for model \"faqs\""}`. 

But if I delete a FAQ and then try to delete it again, I get the proper error message. It's probably not necessary to cover these cases with a custom error message.

Reply:
Anything with 12 characters or more is a valid MongoDB object ID.

Reply:
So you think i should make sure the length is correct and not worry about the deletedCount check?

Reply:
I implemented that, tell me what you think.

Reply:
Both, add the length check before it tries to delete it. That would give a proper error message and avoid a unnecessary query to the database.

Reply:
```
if (_id.length < 12) throw new ReferenceError('ID must be at least 12 characters..');
const deletedFAQ = await FAQ.deleteOne({ _id });
if (deletedFAQ.deletedCount === 0) throw new ReferenceError('There is no FAQ to delete with that ID.');
```

Reply:
You could also add the length check when trying to fetch a FAQ by ID

Reply:
Done!